### PR TITLE
Devices

### PR DESCRIPTION
**Pull request by howbazaar**
Initial read of devices.

Worth noting that there weren't any defined locally in my MAAS, so I had to create one to get the response format.

### CODE REVIEW

File: controller_test.go [hunk 1]
```diff
@@ -131,6 +132,31 @@ func (s *controllerSuite) TestBootResources(c *gc.C) {
...
 }
 
+func (s *controllerSuite) TestDevices(c *gc.C) {
+	controller := s.getController(c)
+	devices, err := controller.Devices(DevicesArgs{})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(devices, gc.HasLen, 1)
+}
+
+func (s *controllerSuite) TestDevicessArgs(c *gc.C) {
```
Comment (mjs) [inline controller_test.go]: ss

File: device.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package gomaasapi
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+	"github.com/juju/version"
```
Comment (mjs) [inline device.go:9]: I hadn't noticed that gomaasapi was using this

**Comment by mjs** [comment]
LGTM

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**